File: backend/schemas/whatsapp.py
from pydantic import BaseModel, validator
from datetime import datetime
from typing import Optional
import re

class WhatsAppGroupBase(BaseModel):
    group_name: str
    is_active: bool = True

class WhatsAppGroupCreate(WhatsAppGroupBase):
    group_invite_link: str
    
    @validator('group_invite_link')
    def extract_invite_id(cls, v):
        match = re.search(r'chat\.whatsapp\.com/([A-Za-z0-9]+)', v)
        if not match:
            raise ValueError('Invalid WhatsApp group invite link format')
        return match.group(1)

class WhatsAppGroupUpdate(BaseModel):
    group_name: Optional[str] = None
    is_active: Optional[bool] = None

class WhatsAppGroupResponse(WhatsAppGroupBase):
    id: int
    group_invite_id: str
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/types/whatsapp.ts
export interface WhatsAppGroup {
  id: number;
  group_name: string;
  group_invite_id: string;
  is_active: boolean;
  created_at: string;
}

export interface WhatsAppGroupCreate {
  group_name: string;
  group_invite_link: string;
  is_active: boolean;
}

export interface WhatsAppGroupUpdate {
  group_name?: string;
  group_invite_link?: string;
  is_active?: boolean;
}

export interface WhatsAppGroupList {
  groups: WhatsAppGroup[];
  total_count: number;
}
